src/network/operator.rs

```diff
@@ -0,0 +1,335 @@
+                }));
+                // Wait until the operator handler is ready.
+                let _ = handler.await;
+            } else {
+                error!("Missing operator address. Please specify an Aleo address in order to operate a pool");
+            }
+        }
+
+        Ok(operator)
+    }
+
+    /// Returns an instance of the operator router.
+    pub fn router(&self) -> OperatorRouter<N> {
+        self.operator_router.clone()
+    }
+
+    /// Returns all the shares in storage.
+    pub fn to_shares(&self) -> Vec<(u32, HashMap<Address<N>, u64>)> {
+        self.state.to_shares()
+    }
+
+    ///
+    /// Performs the given `request` to the operator.
+    /// All requests must go through this `update`, so that a unified view is preserved.
+    ///
+    pub(super) async fn update(&self, request: OperatorRequest<N>) {
+        match request {
+            OperatorRequest::PoolRegister(peer_ip, address) => {
+                if let Some(block_template) = &*self.block_template.read().await {
+                    // Ensure this prover exists in the list first, and retrieve their share difficulty.
+                    let share_difficulty = self
+                        .provers
+                        .write()
+                        .await
+                        .entry(address)
+                        .or_insert((Instant::now(), BASE_SHARE_DIFFICULTY, 0))
+                        .1;
+
+                    // Route a `PoolRequest` to the peer.
+                    let message = Message::PoolRequest(share_difficulty, Data::Object(block_template.clone()));
+                    if let Err(error) = self.peers_router.send(PeersRequest::MessageSend(peer_ip, message)).await {
+                        warn!("[PoolRequest] {}", error);
+                    }
+                } else {
+                    warn!("[PoolRegister] No current block template exists");
+                }
+            }
+            OperatorRequest::PoolResponse(peer_ip, block, prover_address) => {
+                if let Some(block_template) = &*self.block_template.read().await {
+                    // Check that the block is relevant.
+                    if self.ledger_reader.latest_block_height().saturating_add(1) != block.height() {
+                        warn!("[PoolResponse] Peer {} sent a stale candidate block.", peer_ip);
+                        return;
+                    }
+
+                    // Retrieve the coinbase transaction records.
+                    let coinbase_records = match block.to_coinbase_transaction() {
+                        Ok(transaction) => {
+                            // Ensure the owner of the coinbase transaction in the block is the operator address.
+                            let coinbase_records: Vec<Record<N>> =
+                                transaction.to_records().filter(|r| Some(r.owner()) == self.address).collect();
+                            if coinbase_records.len() == 0 {
```

Comment:
```suggestion
                            if coinbase_records.is_empty() {
```
even if `clippy` doesn't complain about this right now, it will in the future